subsys/net/ip/net_pkt.c

```diff
@@ -1774,7 +1774,7 @@ int net_pkt_pull(struct net_pkt *pkt, size_t length)
 		c_op->buf->len -= rem;
 		left -= rem;
 		if (left) {
-			memmove(c_op->pos, c_op->pos+rem, rem);
+			memmove(c_op->pos, c_op->pos+rem, left);
```

Comment:
ouch, indeed.

Reply:
> memmove(...)

Gosh, how expensive this is.

Reply:
Also, a friendly reminder that regression tests were invented exactly for cases like this...